src/UrlMatcher.php

```diff
@@ -231,10 +232,24 @@ private function injectRoutes(): void
             if (!$route->getData('hasMiddlewares')) {
                 continue;
             }
-            $hostPattern = $route->getData('host') ?? '{_host:[a-zA-Z0-9\.\-]*}';
-            $methods = $route->getData('methods');
+
+            if ($route->isMultiHost()) {
+                /** @var string $hosts */
+                $hosts = $route->getData('hosts');
+
+                if (preg_match('~' . RouteParser::VARIABLE_REGEX . '~x', $hosts)) {
+                    throw new RuntimeException('Submasks not allowed with multiple host names.');
```

Comment:
```suggestion
                    throw new RuntimeException('Submasks are not allowed with multiple host names.');
```

Reply:
What are submasks?

Reply:
> What are submasks?

If user set in `Route` host(s) something like `''yii.com", "{user}.yiisoft.com"`

Reply:
Isn't that called "placeholders" in FastRoute? I'd name it like that...